src/main/java/org/trigon/hardware/rev/spark/io/SimulationSparkIO.java

```diff
@@ -63,51 +70,30 @@ public void setInverted(boolean inverted) {
     }
 
     @Override
-    public void enableVoltageCompensation(double voltage) {
-        motor.enableVoltageCompensation(voltage);
-    }
-
-    @Override
-    public void setClosedLoopRampRate(double rampRate) {
-        motor.setClosedLoopRampRate(rampRate);
-    }
-
-    @Override
-    public void setSmartCurrentLimit(int limit) {
-        motor.setSmartCurrentLimit(limit);
-    }
-
-    @Override
-    public void setOpenLoopRampRate(double rampRate) {
-        motor.setOpenLoopRampRate(rampRate);
-    }
-
-    @Override
-    public void setPID(double p, double i, double d) {
-        pidController.setP(p);
-        pidController.setI(i);
-        pidController.setD(d);
-    }
-
-    @Override
-    public void setConversionsFactor(double conversionsFactor) {
-        encoder.setConversionsFactor(conversionsFactor);
+    public void setBrake(boolean brake) {
+        SparkMaxConfig configuration = new SparkMaxConfig();
+        configuration.idleMode(brake ? SparkMaxConfig.IdleMode.kBrake : SparkMaxConfig.IdleMode.kCoast);
+        motor.configure(configuration, SparkBase.ResetMode.kNoResetSafeParameters, SparkBase.PersistMode.kNoPersistParameters);
     }
 
     @Override
-    public void restoreFactoryDefaults() {
-        motor.restoreFactoryDefaults();
+    public void updateSimulation() {
+        if (physicsSimulation == null)
+            return;
+        
+        physicsSimulation.setInputVoltage(motorSimulation.getBusVoltage());
+        physicsSimulation.updateMotor();
+        motorSimulation.iterate(physicsSimulation.getSystemVelocityRotationsPerSecond(), RobotHardwareStats.SUPPLY_VOLTAGE, RobotHardwareStats.getPeriodicTimeSeconds());
```

Comment:
1. simulate the encoder as well 
2. you should probably input the motor vel, not system 
3. not sure if `physicsSimulation.setInputVoltage(motorSimulation.getBusVoltage());` works correctly, test and check

Reply:
if not `physicsSimulation.setInputVoltage(motorSimulation.getBusVoltage());` then what else should I use? And how do I test that specifaclly? It's not working rn, but how do I know if it's because of this or smth else?

Reply:
Sout(motorsimulation.getBusVoltage())

Reply:
Read the docs more